addons/account/models/account_move.py

```diff
@@ -1960,6 +1964,11 @@ def _compute_tax_country_code(self):
         for record in self:
             record.tax_country_code = record.tax_country_id.code
 
+    @api.depends('partner_credit_limit', 'partner_credit')
+    def _compute_partner_credit_warning(self):
+        for record in self:
+            record.partner_credit_warning = record.partner_credit_limit and record.partner_credit > record.partner_credit_limit
```

Comment:
would be nice to replace all of your `for record in self` by stuff like `for move/partner/picking in self`